kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/HazelCastConnection.scala

```diff
@@ -18,39 +18,60 @@
 
 package com.datamountaineer.streamreactor.connect.hazelcast
 
-import java.util.UUID
+import java.net.URI
+import java.util.{Properties, UUID}
+import javax.cache.{Cache, CacheManager, Caching}
 
 import com.datamountaineer.streamreactor.connect.hazelcast.config.{HazelCastConnectionConfig, HazelCastSocketConfig}
+import com.hazelcast.cache.HazelcastCachingProvider
 import com.hazelcast.client.HazelcastClient
 import com.hazelcast.client.config.{ClientConfig, ClientNetworkConfig, SocketOptions}
+import com.hazelcast.client.impl.HazelcastClientProxy
 import com.hazelcast.config.GroupConfig
 import com.hazelcast.core.HazelcastInstance
 
 import scala.collection.JavaConversions._
+
 /**
   * Created by andrew@datamountaineer.com on 10/08/16. 
   * stream-reactor
   */
 object HazelCastConnection {
- def apply(config: HazelCastConnectionConfig): HazelcastInstance = {
+  def buildClient(config: HazelCastConnectionConfig): HazelcastInstance = {
    val clientConfig = new ClientConfig
    val networkConfig = clientConfig.getNetworkConfig
    networkConfig.setAddresses(config.members.toList)
    val groupConfig = new GroupConfig(config.group, config.pass)
    clientConfig.setGroupConfig(groupConfig)
    buildSocketOptions(networkConfig, config.socketConfig)
    clientConfig.setInstanceName(config.group + "-kafka-connect-" + UUID.randomUUID().toString)
-   //val credentials = new UsernamePasswordCredentials("", "")
    HazelcastClient.newHazelcastClient(clientConfig)
- }
+  }
 
- private def buildSocketOptions(clientNetworkConfig: ClientNetworkConfig, socketConfig: HazelCastSocketConfig): SocketOptions = {
+  private def buildSocketOptions(clientNetworkConfig: ClientNetworkConfig, socketConfig: HazelCastSocketConfig): SocketOptions = {
    val socketOptions = clientNetworkConfig.getSocketOptions
    socketOptions.setKeepAlive(socketConfig.keepAlive)
    socketOptions.setTcpNoDelay(socketConfig.tcpNoDelay)
    socketOptions.setReuseAddress(socketConfig.reuseAddress)
    socketOptions.setLingerSeconds(socketConfig.lingerSeconds)
    socketOptions.setBufferSize(socketConfig.bufferSize)
    socketOptions
- }
+  }
+
+  def checkCaches(manager: CacheManager, name: String) : Boolean = {
+    manager.getCacheNames.toList.contains(name)
```

Comment:
I'm actually just removing this because you can't get the cache names like this. It was a check at startu[.